pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,438 @@
+        sprods2 = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms2]
+        present = True
+        for coeff, sprod in zip(coeffs2, sprods2):
+            present = sprod in sprods
+            if not present:
+                break
+            p_index = sprods.index(sprod)
+            if not qml.math.equal(coeff, coeffs[p_index]):
+                present = False
+                break
+            coeffs.pop(p_index)
+            sprods.pop(p_index)
+
+        return present
+
+    return _lc_op(op2)
+
+
+def op_in(ops):
+    """Builds a ``Conditional`` as a boolean function for evaluating
+    if a given operation exist in a specified set of operation.
+
+    Args:
+        ops (str, Operation, Union(list[str, Operation])): string
+            representation or instance of the operation.
+
+    Returns:
+        :class:`OpIn <pennylane.noise.OpIn>`: a boolean function that evaluates to ``True``, if a
+        given operation exist in a specified set of operation(s), irrespective of their wires.
+
+    **Example**
+
+    One may use ``op_in`` with a string representation of the name of the operation:
+
+    >>> cond_func = qml.noise.op_in(["RX", "RY"])
+    >>> cond_func(qml.RX(1.23, wires=[0]))
+    True
+    >>> cond_func(qml.RZ(1.23, wires=[3]))
+    False
+    >>> cond_func([qml.RX(1.23, wires=[1]), qml.RY(4.56, wires=[2])])
+    True
+
+    Additionally, an instance of :class:`Operation <pennylane.operation.Operation>`
+    can also be provided:
+
+    >>> cond_func = qml.noise.op_in([qml.RX(1.0, "dino"), qml.RY(2.0, "rhino")])
+    >>> cond_func(qml.RX(1.23, wires=["eve"]))
+    True
+    >>> cond_func(qml.RY(1.23, wires=["dino"]))
+    True
+    >>> cond_func([qml.RX(1.23, wires=[1]), qml.RZ(4.56, wires=[2])])
+    False
+    """
+    ops = [ops] if not isinstance(ops, (list, tuple, set)) else ops
+    return OpIn(ops)
+
+
+def op_eq(ops):
+    """Builds a ``Conditional`` as a boolean function for evaluating
+    if a given operation is equal to the specified operation.
+
+    Args:
+        ops (str, Operation, Union(list[str, Operation])): string
+            representation or instance of the operation.
+
+    Returns:
+        :class:`OpEq <pennylane.noise.OpEq>`: a boolean function that evaluates to ``True``, if a
+        given operation is equal to the specified set of operation(s), irrespective of their wires.
+
+    **Example**
+
+    One may use ``op_eq`` with a string representation of the name of the operation:
+
+    >>> cond_func = qml.noise.op_eq("RX")
+    >>> cond_func(qml.RX(1.23, wires=[0]))
+    True
+    >>> cond_func(qml.RZ(1.23, wires=[3]))
+    False
+    >>> cond_func("CNOT")
+    False
+
+    Additionally, an instance of :class:`Operation <pennylane.operation.Operation>`
+    can also be provided:
+
+    >>> cond_func = qml.noise.op_eq(qml.RX(1.0, "dino"))
+    >>> cond_func(qml.RX(1.23, wires=["eve"]))
+    True
+    >>> cond_func(qml.RY(1.23, wires=["dino"]))
+    False
```

Comment:
The idea here is to have the conditionals that help assert equivalence between operations based on their types and not attributes (like parameters and wires). I have now more clearly stated it in the docs based on your comment and if a user requires the latter, they should be able to use the already available `qml.equal` with a `BooleanFn` wrapper
```
cond_equal = qml.BooleanFn(lambda ops, **kwargs: (lambda op: qml.equal(op, ops, **kwargs)))
```
And, of course, this could be done without lambda construction as well.